crates/voicevox_core/src/voice_model.rs

```diff
@@ -138,35 +170,21 @@ impl VvmEntryReader {
             .collect();
         Ok(VvmEntryReader::new(reader, entry_map))
     }
-    async fn read_vvm_json<T: DeserializeOwned>(&self, filename: &str) -> Result<T> {
+    async fn read_vvm_json<T: DeserializeOwned>(&self, filename: &str) -> anyhow::Result<T> {
         let bytes = self.read_vvm_entry(filename).await?;
-        serde_json::from_slice(&bytes).map_err(|e| Error::VvmRead {
-            filename: filename.into(),
-            source: Some(e.into()),
-        })
+        serde_json::from_slice(&bytes).map_err(|e| e.into())
     }
 
-    async fn read_vvm_entry(&self, filename: &str) -> Result<Vec<u8>> {
-        let me = self.entry_map.get(filename).ok_or(Error::VvmRead {
-            filename: filename.into(),
-            source: None,
-        })?;
-        let mut manifest_reader =
-            self.reader
-                .entry(me.index)
-                .await
-                .map_err(|_| Error::VvmRead {
-                    filename: filename.into(),
-                    source: None,
-                })?;
+    async fn read_vvm_entry(&self, filename: &str) -> anyhow::Result<Vec<u8>> {
+        let me = self
+            .entry_map
+            .get(filename)
+            .ok_or(anyhow!("Not found in vvm entries: {}", filename))?;
```

Comment:
めちゃくちゃnitなのですが、慣習に則るなら次のどちらかになるかと思います。

```suggestion
        let me = self
            .entry_map
            .get(filename)
            .ok_or_else(|| anyhow!("Not found in vvm entries: {}", filename))?;
```

[`clippy::or_fun_call`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#or_fun_call)というルールが半年ほど前まで`warn`でした(煩すぎたのか今は`allow`です)。

```suggestion
        use anyhow::Context as _;

        let me = self
            .entry_map
            .get(filename)
            .with_context(|| format!("Not found in vvm entries: {}", filename))?;
```

[`anyhow::Context`](https://docs.rs/anyhow/1/anyhow/trait.Context.html)というのがあり、anyhow的にはこちらが推奨されています。

まあ`Option`に対しての`with_context`は好みの範疇になるかと思います。[eyre](https://docs.rs/crate/eyre)というanyhowのフォークがあるのですが、そちらではフォーク主の意向により、`Option`に対しては`.ok_or_else(|| eyre!(…))`という形が推奨されています。

Reply:
なるほどです、勉強になります。

ルールがある場合、そのルールがある理由もあると納得しやすそうに感じました！
ちょっとググって、[`ok_orは常にformat!が実行されてしまうが、ok_or_elseだとNoneのときのみformat!が実行される。`](https://rs.nkmk.me/rust-option-to-result/#h3_0)という理由を発見してすごく納得しました。